sdk/core/core/inc/az_span_builder.h

```diff
@@ -71,6 +71,9 @@ AZ_NODISCARD az_result az_span_builder_append(az_span_builder * const self, az_s
  */
 AZ_NODISCARD az_result az_span_builder_append_byte(az_span_builder * const self, uint8_t const c);
 
+AZ_NODISCARD az_result
+az_span_builder_append_zeros(az_span_builder * const self, size_t const size);
+
 AZ_ACTION_FUNC(az_span_builder_append, az_span_builder, az_span_action)
```

Comment:
this could be an issue since we may want to `#undef` AZ_ACTION_FUNC after usage.

Reply:
I would prefer not to `#undef`. Instead, we should decide where it should live, `public`, `internal` or `private`.

Reply:
so undef is better than just living in one of public internal or private. Because it means the macro is totally unaccessable from outside the relevant header